src/gui/src/renderThread.cpp

```diff
@@ -638,6 +639,112 @@ void RenderThread::drawInstanceNames(QPainter* painter,
   painter->setFont(initial_font);
 }
 
+// Draw the instances' names
+void RenderThread::drawItermLabels(QPainter* painter,
+                                     const std::vector<odb::dbInst*>& insts)
+{
+  if (!viewer_->options_->areInstanceItermsVisible()) {
+    return;
+  }
+  const int minimum_size = viewer_->coarseViewableResolution();
+  const QTransform initial_xfm = painter->transform();
+
+  const QColor text_color = viewer_->options_->instanceNameColor();
+  painter->setPen(QPen(text_color, 0));
+  painter->setBrush(QBrush(text_color));
+
+  const QFont initial_font = painter->font();
+  const QFont text_font = viewer_->options_->instanceNameFont();
+  const QFontMetricsF font_metrics(text_font);
+
+  // minimum pixel height for text (10px)
+  if (font_metrics.ascent() < 10) {
+    // text is too small
+    return;
+  }
+
+  // text should not fill more than 90% of the instance height or width
+  static const float size_limit = 0.9;
+  static const float rotation_limit
+      = 0.85;  // slightly lower to prevent oscillating rotations when zooming
+
+  // limit non-core text to 1/2.0 (50%) of cell height or width
+  static const float non_core_scale_limit = 2.0;
+
+  const qreal scale_adjust = 1.0 / viewer_->pixels_per_dbu_;
+
+  painter->setFont(text_font);
+  for (auto inst : insts) {
+    if (restart_) {
+      break;
+    }
+    dbMaster* master = inst->getMaster();
+    int master_height = master->getHeight();
+    int master_width = master->getHeight();
+
+    if (master_height < minimum_size) {
+      continue;
+    }
+    if (!inst->getMaster()->isCore() && master_width < minimum_size) {
+      // if core cell, just check master height
+      continue;
+    }
```

Comment:
I dont think this would be good to move to another function because of the continue statements. I am not sure if continue can be applied outside of a function, but even if it does work I think it would be less readable.

Reply:
The continue statements would become return statements in the new function.  I think it makes sense to refactor as well as it is a lot of duplication.  If you are uncomfortable with that we can merge it as is and I'll take care of it later.  Let me know how you would like to proceed.

Reply:
Sure I can take care of the refactoring in this PR.